src/iam.tf

```diff
@@ -0,0 +1,105 @@
+resource "aws_iam_role" "sagemaker_endpoint" {
+  name               = "${var.md_metadata.name_prefix}-endpoint"
+  assume_role_policy = data.aws_iam_policy_document.sagemaker_assume_role.json
+}
+
+data "aws_iam_policy_document" "sagemaker_assume_role" {
+  statement {
+    actions = ["sts:AssumeRole"]
+    principals {
+      type        = "Service"
+      identifiers = ["sagemaker.amazonaws.com"]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "sagemaker_endpoint" {
```

Comment:
FYI, not all these bridgecrew checks need to be addressed. Just the higher priority ones, or the ones associated to compliance benchmarks.

Reply:
New commit should address most of this. The only outstanding one without constraints is ECR because of AWS's extensive list of DLCs https://github.com/aws/deep-learning-containers/blob/master/available_images.md which seems to have about 10 different accounts and, I'd imagine there are other publicly available images that may be used.